docs/managing-dependencies.md

```diff
@@ -162,8 +162,16 @@ the `--only` option.
 poetry install --only docs
 ```
 
+{{% warning %}}
+
+Groups provided to `--with`, `--without`, and `--only` should at least be declared in `pyproject.toml`.
+Poetry will raise an **Error** if a group that is not declared is used, letting the user know of a
+potential mistake.
```

Comment:
I believe that this change doesn't need to be mentioned in the docs because it's already the natural expectation. You can't install a group you didn't find. I see this MR as simply catching bugs/typos and I don't think that needs to be called out in the docs.

The developers I've shown the MR to all assumed that this is how Poetry already worked.

Reply:
Then, let's remove this warning box.

Reply:
Good point, removed in  d9aaafe64f8602e6832a2f098275dc9ed0ff7c56